ledger-core/virtual/execute/execute.go

```diff
@@ -771,13 +774,12 @@ func (s *SMExecute) stepSendCallResult(ctx smachine.ExecutionContext) smachine.S
 }
 
 func (s *SMExecute) stepFinishRequest(ctx smachine.ExecutionContext) smachine.StateUpdate {
-
 	if s.migrationHappened {
 		return ctx.Jump(s.stepSendDelegatedRequestFinished)
 	}
 
 	action := func(state *object.SharedState) {
-		state.FinishRequest(s.execution.Isolation, s.execution.Outgoing)
+		state.FinishRequest(s.execution.Isolation, s.execution.Outgoing, s.execution.Result)
```

Comment:
if you pass whole result, then you don't need s.execution.Isolation, you have CallFlags in result